controllers/compaction/reconciler.go

```diff
@@ -366,10 +386,10 @@ func getCompactionJobVolumeMounts(etcd *druidv1alpha1.Etcd, logger logr.Logger)
 		})
 	}
 
-	return vms
+	return vms, nil
 }
 
-func getCompactionJobVolumes(ctx context.Context, cl client.Client, logger logr.Logger, etcd *druidv1alpha1.Etcd) []v1.Volume {
+func getCompactionJobVolumes(ctx context.Context, cl client.Client, logger logr.Logger, etcd *druidv1alpha1.Etcd) ([]v1.Volume, error) {
```

Comment:
You can now remove `logger` from these params as well, similar to what you did for `getCompactionJobVolumeMounts()`

Reply:
`GetHostMountPathFromSecretRef` function used in this method needs the logger